src/OscapScannerRemoteSsh.cpp

```diff
@@ -328,6 +344,29 @@ void OscapScannerRemoteSsh::evaluate()
     signalCompletion(mCancelRequested);
 }
 
+void OscapScannerRemoteSsh::selectError(MessageType& kind, const QString& message)
+{
+    OscapScannerBase::selectError(kind, message);
+    if (mUserIsSudoer)
+    {
+        if (message.contains(QRegExp("^sudo:")))
+        {
+            kind = MSG_ERROR;
+        }
+    }
+
+}
+
+void OscapScannerRemoteSsh::processError(QString& message)
+{
+    OscapScannerBase::processError(message);
+    if (mUserIsSudoer && message.contains(QRegExp("^sudo:")))
+    {
+        message.replace(QRegExp("^sudo:"), "Error invoking sudo on the host:");
+        message += ".\nOnly passwordless sudo setup on the remote host is supported by scap-workbench.";
```

Comment:
As this is a special case what if we also add an additional short description how to do that?
```
Error invoking sudo on the host: a password is required. Only passwordless sudo setup on the remote host is supported by scap-workbench.
To configure a non-privileged user (for example 'oscap-user') to run only the oscap binary as root update the remote machine's 'sudoers' file or add a file like '/etc/sudoers.d/99-oscap-user' with:
# allow oscap-user to run openscap scanner
Defaults!/usr/bin/oscap !requiretty
oscap-user ALL=(root) NOPASSWD: /usr/bin/oscap
```
This will prolong the message but will point a user how to do it. Anyway it is just an idea, feel free to discard it and rather only extend the documentation.